plugin_guide.md

```diff
@@ -63,57 +104,106 @@ class MyModel(ContainerizedModel, MyPluginMethods):
+1. [Register and install your plugin](#registering-your-plugin)
+1. [Update the readme and demo notebook](#update-readme-and-demo-notebook)
+1. [Make your plugin available on PyPI](#upload-to-pypi)
+1. [List your plugin on the eWaterCycle documentation](#listing-your-plugin-on-the-ewatercycle-documentation)
+1. [Make your plugin citeable](https://zenodo.org/account/settings/github/)
+1. Add tests and continuous integration
 
 ### eWaterCycle Forcing
 
-For generating and loading model forcing, eWaterCycle makes use of a Forcing class.
-When generating forcing, ESMValTool recipes are used. This allows for standardized and reproducible forcing.
+For generating and loading model forcing, eWaterCycle makes use of a Forcing
+class. When generating forcing, ESMValTool recipes are used. This allows for
+standardized and reproducible forcing.
 
-If you are making a new model, you can use the GenericForcing, which has a lumped and a gridded version available.
+If you are making a new model, you can use the GenericForcing, which has a
+lumped and a gridded version available.
 
-Otherwise you will have to make your own custom forcing class. For more info on this, see [the eWaterCycle documentation on forcing.](https://ewatercycle.readthedocs.io/en/latest/user_guide.html#Forcing-data).
+If you are using an existing model that deviates from the standard forcing, you
+can make your own custom forcing class. For more info on this, see [the
+eWaterCycle documentation on
+forcing.](https://ewatercycle.readthedocs.io/en/latest/user_guide.html#Forcing-data).
 
+### Registering your plugin
 
-## eWaterCycle plugin entry-point:
-Finally, the model can be registered as a plugin so that eWaterCycle can find it.
-This is done in the `pyproject.toml` file:
+eWaterCycle uses
+[entrypoints](https://packaging.python.org/en/latest/guides/creating-and-discovering-plugins/#using-package-metadata)
+to discover plugins installed in your environment. Making your plugin discoverable is done in the
+`pyproject.toml` file:
 
 ```toml
 # This registers the plugin such that it is discoverable by eWaterCycle
 [project.entry-points."ewatercycle.models"]
 MyModel = "mymodel.ewatercycle_model:MyModel"
 ```
 
-Here you would replace the leaky bucket names with the correct model and class name of your own model.
+Here you should have replaced the leaky bucket names with the correct model and class
+name of your own model.
 
-Now you can do:
+After (re-)installing your model (`pip install -e .`), you should be able to import and run your model in eWaterCycle:
 
 ```py
 from ewatercycle.models import MyModel
 ```
 
-And run the model in eWaterCycle! 🚀
+Well done! 🚀
+
+### Update readme and demo notebook
+
+To help potential users find and use your plugin, it is imperative to have an
+adequate readme and ship an example notebook with the repository that runs a
+simple example case.
 
+A working example notebook is a requirement for listing your plugin on the
+eWaterCycle documentation.
```

Comment:
This repo already has https://github.com/eWaterCycle/ewatercycle-leakybucket/blob/main/demo_containerized_model.ipynb notebook. Should that notebook be made in the working example notebook for your plugin?
```suggestion
eWaterCycle documentation.

You can use the [demo_containerized_model.ipynb](demo_containerized_model.ipynb) notebook as a starting point for a notebook that works with your model.
```